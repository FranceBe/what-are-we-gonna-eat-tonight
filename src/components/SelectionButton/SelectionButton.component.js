import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { spaces, colors, fontSizes } from 'styles/variables'


const buttonStyle = props => `
  background-color: ${props.isActive ? colors.primary_default : colors.neutral_lightest};
  color: ${props.isActive ? colors.neutral_lightest : colors.primary_dark};
  text-decoration: ${props.isActive ? 'underline' : 'none'};
`
export const SingleButton = styled.button`
  ${props => buttonStyle(props)};
  height: ${spaces.large};
  width: ${spaces.xxtraLarge};
  font-size: ${fontSizes.px_14};
  font-weight: 700;
  border: none;
  :focus {
  outline: none;
  }
`
export const ButtonsContainer = styled.div`
  button:first-of-type {
    border-radius: ${spaces.medium} 0 0 ${spaces.medium};
    border: ${spaces.xtraSmall} solid ${colors.primary_default};
  }
   button:last-of-type {
    border-radius: 0 ${spaces.medium} ${spaces.medium} 0;
    border-top: ${spaces.xtraSmall} solid ${colors.primary_default};
    border-right: ${spaces.xtraSmall} solid ${colors.primary_default}; 
    border-bottom: ${spaces.xtraSmall} solid ${colors.primary_default};
  }
`

export class SelectionButton extends React.PureComponent {
  static propTypes = {
    firstOption: PropTypes.string.isRequired,
    secondOption: PropTypes.string.isRequired,
    onSelect: PropTypes.func.isRequired,
    selectedOption: PropTypes.string.isRequired,
  }

  render() {
    return (
      <ButtonsContainer>
        <SingleButton
          onClick={() => this.props.onSelect(this.props.firstOption)}
          isActive={this.props.selectedOption === this.props.firstOption}
        >
          {this.props.firstOption}
        </SingleButton>
        <SingleButton
          onClick={() => this.props.onSelect(this.props.secondOption)}
          isActive={this.props.selectedOption === this.props.secondOption}
        >
          {this.props.secondOption}
        </SingleButton>
      </ButtonsContainer>
    )
  }
}

export default SelectionButton
