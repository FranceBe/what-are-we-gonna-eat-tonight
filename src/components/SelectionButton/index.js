import SelectionButtonComponent,
{ SelectionButton, SingleButton, ButtonsContainer }
  from 'components/SelectionButton/SelectionButton.component'

export { SelectionButton, SingleButton, ButtonsContainer }

export default SelectionButtonComponent
