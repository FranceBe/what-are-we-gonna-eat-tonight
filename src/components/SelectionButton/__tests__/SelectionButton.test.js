import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import SelectionButton,
{ SingleButton, ButtonsContainer }
  from 'components/SelectionButton/SelectionButton.component'

describe('SelectionButton', () => {
  describe('styled components', () => {
    describe('SingleButton', () => {
      it('should match snapshot', () => {
        const SingleButtonTree = renderer.create(<SingleButton />).toJSON()
        expect(SingleButtonTree).toMatchSnapshot()
      })
      describe('given isActive is false', () => {
        it('should have white background, #5F4C44 color and no text-decoration', () => {
          const SingleButtonTree = renderer.create(<SingleButton />).toJSON()
          expect(SingleButtonTree).toHaveStyleRule('background-color', '#FFF')
          expect(SingleButtonTree).toHaveStyleRule('color', '#5F4C44')
          expect(SingleButtonTree).toHaveStyleRule('text-decoration', 'none')
        })
      })
      describe('given isActive is true', () => {
        it('should have #EBA789 background, #FFF color and underline text-decoration', () => {
          const SingleButtonTree = renderer.create(<SingleButton isActive />).toJSON()
          expect(SingleButtonTree).toHaveStyleRule('background-color', '#EBA789')
          expect(SingleButtonTree).toHaveStyleRule('color', '#FFF')
          expect(SingleButtonTree).toHaveStyleRule('text-decoration', 'underline')
        })
      })
    })
    describe('ButtonsContainer', () => {
      it('should match snapshot', () => {
        const ButtonsContainerTree = renderer.create(<ButtonsContainer />).toJSON()
        expect(ButtonsContainerTree).toMatchSnapshot()
      })
    })
  })
  describe('component rendering', () => {
    const defaultProps = {
      firstOption: 'option 1',
      secondOption: 'option 2',
      onSelect: () => {},
      selectedOption: '',
    }
    describe('default props', () => {
      it('should render 1 ButtonsContainer and 2 SingleButton with props', () => {
        const SelectionButtonWrapper = shallow(<SelectionButton {...defaultProps} />)
        expect(SelectionButtonWrapper.find(ButtonsContainer)).toHaveLength(1)
        const SingleButtonWrapper = SelectionButtonWrapper.find(SingleButton)
        expect(SingleButtonWrapper).toHaveLength(2)
        expect(SingleButtonWrapper.at(0).props().isActive).toBe(false)
        expect(SingleButtonWrapper.at(1).props().isActive).toBe(false)
      })
      it('should call onSelect with args when button is clicked', () => {
        const props = {
          onSelect: jest.fn()
        }
        const SelectionButtonWrapper = shallow(<SelectionButton {...defaultProps} {...props} />)
        const SingleButtonWrapper = SelectionButtonWrapper.find(SingleButton)
        SingleButtonWrapper.at(0).props().onClick()
        expect(props.onSelect).toHaveBeenCalledTimes(1)
        expect(props.onSelect).toHaveBeenCalledWith(defaultProps.firstOption)

        SingleButtonWrapper.at(1).props().onClick()
        expect(props.onSelect).toHaveBeenCalledWith(defaultProps.secondOption)
      })
      it('should set isActive to true if props.selectedOption equal the button\'s options', () => {
        const props = {
          selectedOption: 'option 1',
        }
        const SelectionButtonWrapper = shallow(<SelectionButton {...defaultProps} {...props} />)
        const SingleButtonWrapper = SelectionButtonWrapper.find(SingleButton)
        expect(SingleButtonWrapper.at(0).props().isActive).toBe(true)
        expect(SingleButtonWrapper.at(1).props().isActive).toBe(false)
      })
    })
  })
})
