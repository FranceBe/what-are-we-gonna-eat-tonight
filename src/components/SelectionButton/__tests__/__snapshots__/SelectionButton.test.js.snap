// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionButton styled components ButtonsContainer should match snapshot 1`] = `
.c0 button:first-of-type {
  border-radius: 10px 0 0 10px;
  border: 1px solid #EBA789;
}

.c0 button:last-of-type {
  border-radius: 0 10px 10px 0;
  border-top: 1px solid #EBA789;
  border-right: 1px solid #EBA789;
  border-bottom: 1px solid #EBA789;
}

<div
  className="c0"
/>
`;

exports[`SelectionButton styled components SingleButton should match snapshot 1`] = `
.c0 {
  background-color: #FFF;
  color: #5F4C44;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 50px;
  width: 300px;
  font-size: 0.875rem;
  font-weight: 700;
  border: none;
}

.c0:focus {
  outline: none;
}

<button
  className="c0"
/>
`;
